src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -155,7 +166,17 @@ export class MongoCollectionTreeItem implements IAzureParentTreeItem {
 	}
 
 	private async findOne(args?: any): Promise<string> {
-		const result = await this.collection.findOne(args);
+		if (args && args.length > 2) {
+			throw new Error("Too many arguments")
+		}
+		let result;
+		if (args.length === 1) {
+			result = await this.collection.findOne(args[0]);
+		} else if (args.length === 2) {
+			result = await this.collection.findOne(args[0], { fields: args[1] });
+		} else {
+			result = await this.collection.findOne({});
+		}
```

Comment:
This feels like I'm enumerating function overloading. Any suggestions on a succinct way to do this?

Reply:
This get what you have working and then worry about it.